streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java

```diff
@@ -0,0 +1,248 @@
+ * limitations under the License.
+ */
+
+package org.apache.streampark.console.core.task;
+
+import org.apache.streampark.common.enums.ClusterState;
+import org.apache.streampark.common.enums.ExecutionMode;
+import org.apache.streampark.common.util.HttpClientUtils;
+import org.apache.streampark.common.util.ThreadUtils;
+import org.apache.streampark.common.util.YarnUtils;
+import org.apache.streampark.console.base.util.JacksonUtils;
+import org.apache.streampark.console.core.entity.FlinkCluster;
+import org.apache.streampark.console.core.metrics.flink.Overview;
+import org.apache.streampark.console.core.metrics.yarn.YarnAppInfo;
+import org.apache.streampark.console.core.service.FlinkClusterService;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.config.RequestConfig;
+
+import com.baomidou.mybatisplus.core.conditions.update.LambdaUpdateWrapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/** This implementation is currently used for tracing Cluster on yarn,remote,K8s mode */
+@Slf4j
+@Component
+public class FlinkClusterWatcher {
+
+  @Autowired private FlinkClusterService flinkClusterService;
+
+  private Long lastWatcheringTime = null;
+
+  // Track interval  every 30 seconds
+  private static final Long WATCHER_INTERVAL = 1000L * 30;
+
+  /** Watcher cluster lists */
+  private static final Map<Long, FlinkCluster> WATCHER_CLUSTERS = new ConcurrentHashMap<>(0);
+
+  /** Thread pool for processing status monitoring for each cluster */
+  private static final ExecutorService EXECUTOR =
+      new ThreadPoolExecutor(
+          Runtime.getRuntime().availableProcessors() * 5,
+          Runtime.getRuntime().availableProcessors() * 10,
+          60L,
+          TimeUnit.SECONDS,
+          new LinkedBlockingQueue<>(1024),
+          ThreadUtils.threadFactory("flink-cluster-watching-executor"));
+
+  /** Initialize cluster cache */
+  @PostConstruct
+  private void init() {
+    WATCHER_CLUSTERS.clear();
+    List<FlinkCluster> flinkClusters = flinkClusterService.list();
+    flinkClusters.forEach(cluster -> WATCHER_CLUSTERS.put(cluster.getId(), cluster));
+  }
+
+  /** flinkcluster persistent */
+  @PreDestroy
+  private void stop() {
+    // TODO: flinkcluster persistent
+  }
+
+  @Scheduled(fixedDelay = 1000)
+  private void start() {
+    if (lastWatcheringTime == null) {
+      watcher();
+    } else if (System.currentTimeMillis() - lastWatcheringTime >= WATCHER_INTERVAL) {
```

Comment:
```suggestion
   if (System.currentTimeMillis() - lastWatcheringTime >= WATCHER_INTERVAL) {
```
It could be adjusted due to `anchorA`.

Reply:
> It could be adjusted due to `anchorA`.

I will modify the default values later.